telescope/selector_test.py

```diff
@@ -126,8 +127,133 @@ def testValidInput_v1dot1_Simple(self):
+            "file_format_version": 1.1,
+            "duration": "30d",
+            "metrics": ["average_rtt"],
+            "ip_translation":{
+                "strategy":"maxmind",
+                "params":{
+                    "db_snapshots":["2014-08-04"]
+                }
+            },
+            "sites": ["lga02"],
+            "start_times": ["2014-02-01T00:00:00Z"]
+        }"""
+    selector_expected = selector.Selector()
+    selector_expected.start_time = utils.make_datetime_utc_aware(datetime.datetime(2014, 2, 1))
+    selector_expected.duration = 30 * 24 * 60 * 60
+    selector_expected.metric = 'average_rtt'
+    selector_expected.ip_translation_spec = (
+         iptranslation.IPTranslationStrategySpec('maxmind',
+                                                 {'db_snapshots': ['2014-08-04']}))
+    selector_expected.site = 'lga02'
+    self.assertParsedSingleSelectorMatches(selector_expected, selector_file_contents)
+
+    selector_file_contents = """{
+            "file_format_version": 1.1,
+            "duration": "30d",
+            "metrics": ["average_rtt"],
+            "ip_translation":{
+                "strategy":"maxmind",
+                "params":{
+                    "db_snapshots":["2014-08-04"]
+                }
+            },
+            "client_providers": ["comcast"],
+            "start_times": ["2014-02-01T00:00:00Z"]
+        }"""
+    selector_expected = selector.Selector()
+    selector_expected.start_time = utils.make_datetime_utc_aware(datetime.datetime(2014, 2, 1))
+    selector_expected.duration = 30 * 24 * 60 * 60
+    selector_expected.metric = 'average_rtt'
+    selector_expected.ip_translation_spec = (
+         iptranslation.IPTranslationStrategySpec('maxmind',
+                                                 {'db_snapshots': ['2014-08-04']}))
+    selector_expected.client_provider = 'comcast'
+    self.assertParsedSingleSelectorMatches(selector_expected, selector_file_contents)
+
+    selector_file_contents = """{
+            "file_format_version": 1.1,
+            "duration": "30d",
+            "metrics": ["average_rtt"],
+            "ip_translation":{
+                "strategy":"maxmind",
+                "params":{
+                    "db_snapshots":["2014-08-04"]
+                }
+            },
+            "client_countries": ["us"],
+            "start_times": ["2014-02-01T00:00:00Z"]
+        }"""
+    selector_expected = selector.Selector()
+    selector_expected.start_time = utils.make_datetime_utc_aware(datetime.datetime(2014, 2, 1))
+    selector_expected.duration = 30 * 24 * 60 * 60
+    selector_expected.metric = 'average_rtt'
+    selector_expected.ip_translation_spec = (
+         iptranslation.IPTranslationStrategySpec('maxmind',
+                                                 {'db_snapshots': ['2014-08-04']}))
+    selector_expected.client_country = 'us'
+    self.assertParsedSingleSelectorMatches(selector_expected, selector_file_contents)
+
+
```

Comment:
Can we add a test case where one of the optional parameters is an empty list? e.g. `metrics: []`

Such a file seems to adhere to our spec, but I think this breaks our code now. I think we should treat an empty list the same as if they omitted the field.